### PR TITLE
Add update sleep cycle feature

### PR DESCRIPTION
**Pull request by OKW32**

### CODE REVIEW

File: src/main/java/commands/sleepcommands/UpdateSleepCommand.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if (userCommand.length != 2) {
+            throw new SleepException("Please use proper format: " + System.lineSeparator()
+                    + "sleep update <date> /new <hours>");
+        }
+        if (userCommand[0].isBlank()) {
+            throw new SleepException("Key in non-empty date");
+        }
+        if (userCommand[1].isBlank()) {
+            throw new SleepException("Key in non-empty number of hours to update to");
+        }
```
Comment (Daviancold) [inline src/main/java/commands/sleepcommands/UpdateSleepCommand.java:23]: Can consider adding a comment to help the user better see what strings should be inside the userCommand Array

File: src/main/java/commands/sleepcommands/UpdateSleepCommand.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        try {
+            this.sleepTracker = sleepTracker;
+            this.date = userCommand[0].trim();
+            this.hours = Double.parseDouble(userCommand[1].trim());
```
Comment (Daviancold) [inline src/main/java/commands/sleepcommands/UpdateSleepCommand.java:27]: Nothing wrong with this code, but the "this" identifier is not needed in this case as the parameters do not have the same name as the local attributes

**Review by Daviancold — APPROVED** [review]
Overall clean code, can consider adding more documentation for the updateSleepCommand code!

File: src/main/java/commands/sleepcommands/UpdateSleepCommand.java (hunk 1 above)
Comment (OKW32) [reply src/main/java/commands/sleepcommands/UpdateSleepCommand.java:23]: There is a line before this that ensures that will prompt users to use proper format when improper format is keyed in. This line of code will throw the exceptions only when they use proper format, but empty parameters. For e.g sleep update /new.

File: src/main/java/commands/sleepcommands/UpdateSleepCommand.java (hunk 2 above)
Comment (OKW32) [reply src/main/java/commands/sleepcommands/UpdateSleepCommand.java:27]: Noted will remove them.

**Review by Daviancold — APPROVED** [review]

**Pull request merged**